### PR TITLE
fix: 'inferenceConfig.stopSequences' failed to satisfy constraint: Me…

### PR DESCRIPTION
**Pull request by malaki12003**
…mber must have length less than or equal to 4 #2538

## Description

According to AWS Bedrock's requirements, we need to define at least four stop sequences, as outlined in the AWS documentation: [API Inference Configuration](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_agent_InferenceConfiguration.html). As a result, I have limited the stop sequences to a maximum of four.

## Checklist

- [X] The base branch of this PR is `dev`, rather than `main`
- [X] The relevant docs, if any, have been updated or created

## Testing

Auto-completion for Bedrock is working.

### CODE REVIEW

File: core/llm/llms/Bedrock.ts [hunk 1]
```diff
@@ -111,7 +111,15 @@ class Bedrock extends BaseLLM {
         maxTokens: options.maxTokens,
         temperature: options.temperature,
         topP: options.topP,
-        stopSequences: options.stop?.filter((stop) => stop.trim() !== ""),
+        // TODO: The current approach selects the first 4 items from the list to comply with Bedrock's requirement
+        // of having at least 4 stop sequences, as per the AWS documentation:
```
Comment (tahouse) [inline core/llm/llms/Bedrock.ts]: Hi, minor nit: this should say "at most 4 stop sequences" not "at least". Documentation specifies: "Minimum number of 0 items. Maximum number of 4 items."

The slice logic below looks fine; just the comment is misleading.

File: core/llm/llms/Bedrock.ts [hunk 2]
```diff
@@ -111,7 +111,15 @@ class Bedrock extends BaseLLM {
         maxTokens: options.maxTokens,
         temperature: options.temperature,
         topP: options.topP,
-        stopSequences: options.stop?.filter((stop) => stop.trim() !== ""),
+        // TODO: The current approach selects the first 4 items from the list to comply with Bedrock's requirement
+        // of having at most 4 stop sequences, as per the AWS documentation:
+        // https://docs.aws.amazon.com/bedrock/latest/APIReference/API_agent_InferenceConfiguration.html
+        // However, it might be better to implement a strategy that dynamically selects the most appropriate stop sequences
+        // based on the context.
+        // TODO: Additionally, consider implementing a global exception handler for the providers to give users clearer feedback.
```
Comment (Patrick-Erichsen) [inline core/llm/llms/Bedrock.ts:119]: Good feedback here but I don't think we want to include this in the source code.

**Review by Patrick-Erichsen — CHANGES_REQUESTED** [review]

File: core/llm/llms/Bedrock.ts (hunk 2 above)
Comment (Patrick-Erichsen) [reply core/llm/llms/Bedrock.ts:119]: Specifically referencing the second "TODO" here, the first on is helpful.

Comment (Patrick-Erichsen) [reply core/llm/llms/Bedrock.ts:119]: Went ahead and just merged this to get the bugfix out ASAP.

**Comment by sestinj** [comment]
Hi @malaki12003, yesterday we shared [some updates](https://discordapp.com/channels/1108621136150929458/1131313996750917835/1371996789967687783) with our contributors about how we're aiming to improve the contribution process. Part of this included the addition of a [Contributor License Agreement](https://github.com/continuedev/continue/blob/main/docs/docs/CLA.md) (CLA) to protect both contributors and the project. We're reaching out to ask that previous contributors sign it.

Could you please take a moment to sign, or if you have any questions send me a message? (either here or nate@continue.dev would work)

To do so, you just need to **post a comment below with the following text:**<br/>
```
I have read the CLA Document and I hereby sign the CLA
```

❤️ Thank you for the work you've done on Continue, and let me know if you have any suggestions on how we can make the project even better!

**Comment by Ezeikelthe1st** [comment]
is this autocompleter for bedrock learning or minecraft bedrock

**Pull request merged**